automatminer/automl/adaptors.py

```diff
@@ -226,6 +228,129 @@ def predict(self, df, target):
+        self.learning_rate = learning_rate
+        self.stop_early = stop_early
+        self._backend = None
+        self.kwargs = kwargs
+        self._features = None
+        self._logger = self.get_logger(logger)
+        self.is_fit = False
+
+    @set_fitted
+    def fit(self, df, target, **fit_kwargs):
+        """
+        Train a neural network by fitting on a dataframe.
+
+        Args:
+            df (pandas.DataFrame): The df to be used for training.
+            target (str): The key used to identify the machine learning target.
+            **fit_kwargs: Keyword arguments to be passed to the neural network's fit method.
+                These arguments must be valid arguments to the Keras neural network fit method.
+
+        Returns:
+            NeuralNetworkAdaptor(self)
+
+        """
+        y = df[target]
+        X = df.drop(columns=target)
+
+        # Determine learning type based on whether classification or regression
+        self.mode = regression_or_classification(df[target])
+        if self.mode == "classification" or self.mode == "regression":
+            self._features = df.drop(columns=target).columns.tolist()
+            self._ml_data = {"X": X, "y": y}
+            self.fitted_target = target
+            print(self._logger)
+            self._logger.info("Neural network fitting started.")
+            self._backend = NnWrapper(self.hidden_layer_sizes, self.dropout, self.show_accuracy, self.batch_spec, self.activation, self.input_noise, self.use_maxout, self.use_maxnorm, self.learning_rate, self.stop_early, self.mode)
+            self._backend.fit(X, y, **fit_kwargs)
+            self._logger.info("Neural network fitting finished.")
+        else:
+            raise ValueError("Learning type {} not recognized as a valid mode "
+                             "for {}".format(self.mode, self.__class__.__name__))
+        return self
+
+    @check_fitted
+    def predict(self, df, target):
+        """
+        Predict the target property of materials given a df of features.
+
+        The predictions are appended to the dataframe in a column called:
+            "{target} predicted"
+
+        Args:
+            df (pandas.DataFrame): Contains all features needed for ML (i.e.,
+                all features contained in the training dataframe.
+            target (str): The property to be predicted. Should match the target
+                used for fitting. May or may not be present in the argument
+                dataframe.
+
+        Returns:
+            (pandas.DataFrame): The argument dataframe plus a column containing
+                the predictions of the target.
+
+        """
+        if target != self.fitted_target:
+            raise AutomatminerErrorError("Argument dataframe target {} is different from"
+                                " the fitted dataframe target! {}"
+                                "".format(target, self.fitted_target))
+        elif not self.is_fit:
```

Comment:
This is_fit check here is superflous. Check @check_fitted decorator for details. We'll be removing these similar lines from TPOTAdaptor